packages/gatsby/src/query/error-parser.ts

```diff
@@ -100,12 +139,12 @@ const errorParser = ({
   let structured
 
   for (const { regex, cb } of handlers) {
-    const matched = message.match(regex)
+    const matched = message?.match(regex)
     if (matched) {
       structured = {
-        ...(filePath && { filePath }),
-        ...(location && { location }),
         ...cb(matched),
+        ...{ location },
```

Comment:
TS was complaining about `...(location && { location })` so I dropped the check. It'll return `undefined` now. Which is fine as we check for truthy values on that further down the line.